vignettes/b-advanced-ggplot.Rmd

```diff
@@ -0,0 +1,272 @@
+
+```{r message = FALSE}
+# Load libraries
+library(rphylopic)
+library(ggplot2)
+library(phytools)
+# Get vertebrate phylogeny and data
+data(vertebrate.tree)
+```
+
+We can use a vectorized version of the `get_uuid()` function to retrieve UUID values for all of the species at once. However, just in case we get an error, we wrap the `get_uuid()` call in a `tryCatch()` call. This way, we should get either a UUID or `NA` for each species: 
+
+```{r}
+# Make a data.frame for the PhyloPic names
+vertebrate.data <- data.frame(species = vertebrate.tree$tip.label, uuid = NA)
+# Try to get PhyloPic UUIDs for the species names
+vertebrate.data$uuid <- sapply(vertebrate.tree$tip.label,
+                               function(x) tryCatch(get_uuid(x),
+                                                    error = function(e) NA))
+vertebrate.data
+```
+
+Oh no, we weren't able to find a silhouette for *Myotis lucifugus* (little brown bat)! Good thing we used `tryCatch()`! Given the coarse resolution of this phylogeny, we can just grab a silhouette for the subfamily (Vespertilioninae):
+
+```{r}
+vertebrate.data$uuid[vertebrate.data$species == "Myotis_lucifugus"] <-
+  get_uuid("Vespertilioninae")
+```
+
+I'm also not a huge fan of the boar picture. Let's choose an alternative with `pick_phylopic()`.
+
+```{r eval = FALSE}
+# Pick a different boar image; we'll pick #2
+boar_svg <- pick_phylopic("Sus scrofa", view = 5)
+# Extract the UUID
+vertebrate.data$uuid[vertebrate.data$species == "Sus_scrofa"] <-
+  attr(boar_svg, "uuid")
+```
+
+```{r echo = FALSE}
+vertebrate.data$uuid[vertebrate.data$species == "Sus_scrofa"] <- "87047da1-b40e-4b31-8492-4db262f129f5"
+```
+
+Now that we've got our phylogeny and UUIDs, we could go ahead and create our figure. However, time for a quick aside. The time required for `geom_phylopic()` and the other **rphylopic** visualization functions scales with the number of *unique* names/UUIDs, not the number of plotted silhouettes. Therefore, if you are plotting a lot of *different* silhouettes, these functions can take quite a long time to poll PhyloPic for each unique name, download the silhouettes, and convert them to be added to the plot. If you plan to use the same silhouettes for multiple figures, we strongly suggest that you poll PhyloPic yourself using `get_phylopic()` or `pick_phylopic()`, save the silhouettes to your R environment, and then these use image objects in the visualization functions (with the `img` argument/aesthetic). Following this advice, let's get image objects for these 11 species before we make our figure. Note that, since we've used `get_uuid()` to get these 11 UUIDs, we know that they are valid, so we don't need to catch any errors this time.
+
+```{r}
+vertebrate.data$svg <- lapply(vertebrate.data$uuid, get_phylopic)
+```
+
+Now let's go ahead and plot our phylogeny with the [ggtree](https://www.bioconductor.org/packages/ggtree) package:
+
+```{r message = FALSE}
+library(ggtree)
```

Comment:
Obviously this isn't currently available on CRAN for the most recent versions of R. Is this potentially problematic for the vignette building?